src/main/scala/edu/twitter/model/evaluation/ModelEvaluator.scala

```diff
@@ -0,0 +1,148 @@
+
+/** Representation of the records used for training
+  * and testing the model. */
+case class Record(tweetText: String, actualLabel: Double)
+
+/** Grouping of the evaluation necessary fields
+  * these fields are used for evaluating the training error. */
+case class EvaluationFields(happyCorrect: Int, happyTotal: Int, sadCorrect: Int, sadTotal: Int) {
+
+  /**
+    * Combining two evaluation instances.
+    *
+    * @param o instance to combine with this one.
+    * @return a new instance resulting of the combination of
+    *         both instances.
+    */
+  def +(o: EvaluationFields): EvaluationFields = o match {
+    case EvaluationFields(oHappyCorrect, oHappyTotal, oSadCorrect, oSadTotal) =>
+      EvaluationFields(
+        happyCorrect + oHappyCorrect,
+        happyTotal + oHappyTotal,
+        sadCorrect + oSadCorrect,
+        sadTotal + oSadTotal
+      )
+  }
+}
+
+/** Representation of a training tweet that has been classified
+  * by the model, it holds both the actual and model labels. */
+case class EvaluatedTrainingTweet(actualLabel: Double, modelPrediction: Double, tweetText: String)
+
+/**
+  * Responsible for evaluating a model based on a given
+  * testing data. this class will print training analysis
+  * in the console and persist the labeled data in `Elasticsearch`
+  * for further visualization.
+  *
+  * @param sc spark context.
+  */
+class ModelEvaluator(sc: SparkContext) {
+  /**
+    * Given a model will evaluate it based on
+    * training data and log evaluation analysis.
+    *
+    * @param model target model for evaluation.
+    */
+  def evaluate(model: GenericModel): Unit = {
+    val tweetsLoader = new TweetsLoader(sc)
+    val evaluation = evaluateData(model, tweetsLoader.getTweetsDataSet())
+    performEvaluationAnalysis(evaluation)
+  }
+
+  /**
+    * Given a model and its training, testing data. this method
+    * will measure how will the model will perform.
+    *
+    * @param model             target model for evaluation
+    * @param trainingRecords   model's training data
+    * @param validationRecords model's validation data
+    */
+  def evaluate(model: GenericModel, trainingRecords: RDD[Record], validationRecords: RDD[Record]): Unit = {
+    val trainingEvaluation = evaluateRecords(model, trainingRecords)
+    val validationEvaluation = evaluateRecords(model, validationRecords)
+    performEvaluationAnalysis(trainingEvaluation, "Training")
+    performEvaluationAnalysis(validationEvaluation)
+  }
+
+  /**
+    * Given a model will evaluate it based on
+    * training data, log evaluation analysis and
+    * persist the labeled records in `Elasticsearch`
+    *
+    * @param model target model for evaluation
+    */
+  def evaluateAndPersist(model: GenericModel): Unit = {
+    val tweetsLoader = new TweetsLoader(sc)
+    val evaluation = evaluateData(model, tweetsLoader.getTweetsDataSet())
+    performEvaluationAnalysis(evaluation)
+    EsSpark.saveToEs(evaluation, "training-analysis")
```

Comment:
can you change the index name of the elastic search to be something like "index name/ mapping name"
like "training/analysis".
it will help if we want to add a schema to the index later.